.github/workflows/link-checker.yml

```diff
@@ -1,14 +1,26 @@
 name: Link Checker
+
 on:
-  push:
-    branches: [ main ]
   pull_request:
-    branches: [ main ]
+    types: [opened, reopened]
+  push:
+    branches:
+      - 'main'
+      - '[1-9]+.[0-9x]+'
+      - '[1-9]+.[0-9x]+.[0-9x]+'
+    paths:
+      - '**/*.pdf'
+      - '**/*.pnj'
+      - '**/*.jpg'
+      - '**/*.json'
+      - '**/*.md'
+      - '**/*.txt'
+      - '**/*.rst'
```

Comment:
if the workflow only triggers when we modify text, we cannot tell if any existing links became dead in other PRs?

also why `'**/*.jpg'` and what is .pnj?

Reply:
Do we have links that we need to check outside of text/docs? I'm ready to change the trigger criteria.
I included there pictures (`png`, `jpg`), because there is a number of documentation files which have pictures (links to picture files) ([sample](https://raw.githubusercontent.com/opensearch-project/sql/2.x/bi-connectors/TableauConnector/tableau_support.md)). This workflow checks these links as well.

`.pnj` is a mistake - fixed in 489f2adb. Thank you.

Reply:
For example we have a link to `docs.opensearch.org` in some README.md, next day that link becomes `opensearch.org/docs` and `docs.opensearch.org` is no longer valid

But we will only find out our README.md is out of date when we modify a file in this list (which doesn't happen as frequent), so I'm saying should this run regardless of what changed?

Reply:
I understand. Agreed, I will change that.

Reply:
Fixed in 63c51377.